### DISCUSSION TITLE
canyoublockit.com vs. AdGuardHome v0.104.3

### DISCUSSION THREAD

**Discussion by tackynugget** [AdguardTeam/AdGuardHome#2764] [Q&A] [answered]
* **Version of AdGuard Home server:**
  * v0.104.3
* **How did you install AdGuard Home:**
  * Built from source + docker following these steps: https://github.com/boostchicken/udm-utilities/tree/master/AdguardHome
* **How did you setup DNS configuration:**
  * Forced_intfc settings in 10-dns.sh script
* **If it's a router or IoT, please write device model:**
  * Ubiquiti Dream Machine Pro
* **CPU architecture:**
  * ARM (I think)
* **Operating system and version:**
  * Controller 6.0.43 / Firmware 1.8.6

### Expected Behavior
Block ads and pass ad-blocking tests.

### Actual Behavior
Generally blocks ads, but does not perform well against canyoublockit.com and other ad-blocking test sites.

### Screenshots

<details><summary>Screenshot:</summary>

[Screen Shot 2021-02-24 at 10 44 00 PM](https://user-images.githubusercontent.com/12820826/109104594-e076a580-76f1-11eb-8536-883829662ce8.png)

</details>

### Additional Information
AdGuardHome + AdGuard Chrome Browser extension seems to pass the canyoublockit.com test. But AdGuardHome on its own does not.

**Comment by bbartlomiej** [comment]
This depends heavily on the blocklists you use. I tried your canyoublockit.com link and on both basic and extreme test I don't see advertisements. AdBlockHome + uBlock Origin

**Comment by ghost** [comment]
On second test "eXtreme Test" for DNS based AdBlockers like AGH it works fine for me.

No ads and banners, I use AGH SDNS, Adaway, Spam404 and NoCoin list.

**Comment by tackynugget** [comment]
Thanks for the feedback.

@bbartlomiej I couldn't find a ublock origin list. It looks like a similar utility to AdGuardHome. Is there a ublock origin list somewhere that I'm missing?

@uservictor I appear to be using the same lists except for SDNS.
[Screen Shot 2021-02-24 at 10 44 00 PM](https://user-images.githubusercontent.com/12820826/109104594-e076a580-76f1-11eb-8536-883829662ce8.png)

**Comment by ghost** [comment]
SDNS is the default AGH DNS list, on the "first" test like your picture i see the same pictures.
The other test doesnt show any ad.

You can see ads that are not in the list you use, this has nothing to do with AGH, as its function is "just" to check if an requested url is on the blacklist that you use or not.
If there is an match it will be blocked.

**Comment by tackynugget** [comment]
@uservictor Understood.

Is there a Ublock origin list or a better way to consolidate these lists?

I think I read about blocking Google DNS at the firewall level to stop those ads too.

**Reply by bbartlomiej** [reply]
@OverengineeredNetwork  you need to better understand what you're doing. Blocking Google DNS will not give you anything good and it may cause outages unless you redirect this DNS traffic to your own trusted DNS. uBlock Origin is not a list, it's a browser addon which also blocks ads. You need to do your research and find the lists that suit your needs. The more lists you add the more false positives you will get. You will have to do some explicit whitelisting in AdGuardHome as well.

See my lists and whitelists below:
```filters:
- enabled: true
  url: https://adguardteam.github.io/AdGuardSDNSFilter/Filters/filter.txt
  name: AdGuard DNS filter
  id: 1
- enabled: true
  url: https://www.malwaredomainlist.com/hostslist/hosts.txt
  name: MalwareDomainList.com Hosts List
  id: 4
- enabled: true
  url: https://raw.githubusercontent.com/Perflyst/PiHoleBlocklist/master/SmartTV-AGH.txt
  name: Perflyst and Dandelion Sprout's Smart-TV Blocklist
  id: 1599082991
- enabled: true
  url: https://pgl.yoyo.org/adservers/serverlist.php?hostformat=adblockplus&showintro=1&mimetype=plaintext
  name: Peter Lowe's List
  id: 1599082993
- enabled: true
  url: https://raw.githubusercontent.com/durablenapkin/scamblocklist/master/adguard.txt
  name: Scam Blocklist by DurableNapkin
  id: 1599082994
- enabled: true
  url: https://raw.githubusercontent.com/Spam404/lists/master/main-blacklist.txt
  name: Spam404
  id: 1599082995
- enabled: true
  url: https://raw.githubusercontent.com/mitchellkrogza/The-Big-List-of-Hacked-Malware-Web-Sites/master/hacked-domains.list
  name: The Big List of Hacked Malware Web Sites
  id: 1599082997
- enabled: true
  url: https://raw.githubusercontent.com/MajkiIT/polish-ads-filter/master/polish-pihole-filters/hostfile.txt
  name: 'POL: Polish filters for Pi hole'
  id: 1599082998
- enabled: true
  url: https://s3.amazonaws.com/lists.disconnect.me/simple_ad.txt
  name: Simple Ad
  id: 1599082999
- enabled: true
  url: https://s3.amazonaws.com/lists.disconnect.me/simple_tracking.txt
  name: Simple Tracking
  id: 1599083000
- enabled: true
  url: https://someonewhocares.org/hosts/zero/hosts
  name: Dan Pollock's List
  id: 1599083001
- enabled: true
  url: https://adaway.org/hosts.txt
  name: AdAway Default Blocklist
  id: 1606508407
- enabled: true
  url: https://raw.githubusercontent.com/hoshsadiq/adblock-nocoin-list/master/hosts.txt
  name: NoCoin Filter List
  id: 1606508408
- enabled: true
  url: https://paulgb.github.io/BarbBlock/blacklists/hosts-file.txt
  name: BarbBlock
  id: 1606508409
- enabled: true
  url: https://blocklistproject.github.io/Lists/crypto.txt
  name: blocklistproject-crypto
  id: 1606508412
- enabled: true
  url: https://blocklistproject.github.io/Lists/drugs.txt
  name: blocklistproject-drugs
  id: 1606508413
- enabled: true
  url: https://blocklistproject.github.io/Lists/fraud.txt
  name: blocklistproject-fraud
  id: 1606508415
- enabled: true
  url: https://blocklistproject.github.io/Lists/gambling.txt
  name: blocklistproject-gambling
  id: 1606508416
- enabled: true
  url: https://blocklistproject.github.io/Lists/phishing.txt
  name: blocklistproject-phishing
  id: 1606508418
- enabled: true
  url: https://blocklistproject.github.io/Lists/ransomware.txt
  name: blocklistproject-ransomware
  id: 1606508421
- enabled: true
  url: https://blocklistproject.github.io/Lists/scam.txt
  name: blocklistproject-scam
  id: 1606508423
- enabled: true
  url: https://dbl.oisd.nl/
  name: https://dbl.oisd.nl/
  id: 1608459138
- enabled: true
  url: https://raw.githubusercontent.com/Sekhan/TheGreatWall/master/TheGreatWall.txt
  name: The Great Wall
  id: 1608459139
- enabled: true
  url: https://www.github.developerdan.com/hosts/lists/hate-and-junk-extended.txt
  name: Hate and Junk
  id: 1608459140
- enabled: true
  url: https://www.github.developerdan.com/hosts/lists/ads-and-tracking-extended.txt
  name: Ads and Tracking
  id: 1608459142
- enabled: true
  url: https://v.firebog.net/hosts/Easylist.txt
  name: EasyList
  id: 1608459143
- enabled: true
  url: https://v.firebog.net/hosts/AdguardDNS.txt
  name: AdGuard DNS
  id: 1608459144
- enabled: true
  url: https://v.firebog.net/hosts/Admiral.txt
  name: Admiral
  id: 1608459145
- enabled: true
  url: https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts
  name: StevenBlack
  id: 1609090176
whitelist_filters: []
user_rules:
- /^ad([sxv]?[0-9]*|system)[_.-]([^.[:space:]]+\.){1,}|[_.-]ad([sxv]?[0-9]*|system)[_.-]/
- /^(.+[_.-])?adse?rv(er?|ice)?s?[0-9]*[_.-]/
- /^(.+[_.-])?telemetry[_.-]/
- /^adim(age|g)s?[0-9]*[_.-]/
- /^adtrack(er|ing)?[0-9]*[_.-]/
- /^advert(s|is(ing|ements?))?[0-9]*[_.-]/
- /^aff(iliat(es?|ion))?[_.-]/
- /^analytics?[_.-]/
- /^banners?[_.-]/
- /^beacons?[0-9]*[_.-]/
- /^count(ers?)?[0-9]*[_.-]/
- /^mads\./
- /^pixels?[-.]/
- /^stat(s|istics){1}[0-9]*[_.-]/
- '@@||1drv.ms^$important'
- '@@||tvn.adocean.pl^$important'
- '@@||tvn.hit.gemius.pl^$important'
- '@@||usage-eu-fy2018.myhomescreen.tv^$important'
- '@@||s372.meetrics.net^$important'
- '@@||www.interia.pl^$important'
- '@@||xpu.samsungelectronics.com^$important'
- '@@||upu.samsungelectronics.com^$important'
- '@@||lcprd1.samsungcloudsolution.net^$important'
- '@@||lcprd2.samsungcloudsolution.net^$important'
- '@@||notice.samsungcloudsolution.com^$important'
- '@@||fkp.samsungcloudsolution.com^$important'
- '@@||configprd.samsungcloudsolution.net^$important'
- '@@||www.samsungrm.net^$important'
- '@@||sas.samsungcloudsolution.com^$important'
- '@@||prov.samsungcloudsolution.com^$important'
- '@@||fonts.gstatic.com^$important'
- '@@||af.opera.com^$important'
- '@@||dns-family.adguard.com^$important'
- '@@||concierge.analytics.console.aws.a2z.com^$important'
- '@@||foreverliving.com^$important'
- '@@||whispersystems.org^$important'
- '@@||signal.org^$important'
- '@@||foldingathome.org^$important'
- '@@||binance.com^$important'
```

**Reply by tackynugget** [reply]
@bbartlomiej Thank you for the sample lists.

I do understand how this works and have added several supplemental lists. 

One thing I should have mentioned earlier was that I **_can_** reproduce the desired state when I **_ALSO_** activate the AdGuard AdBlocker Chrome browser extension. I reviewed the lists the extension used to block the canyoublockit images (those images **are** the ads, btw--many other forum posts here seem to not understand that), but I couldn't find any difference between the lists the extension used and the lists AdGuardHome uses. Ideally, I'd like my network to enjoy the same level of ad blocking that the chrome extension provides. 

I assumed ublock was another list, as it seemed weird that the solution recommended in other canyoublockit.com threads would involve pulling in another, non-AdGuard solution.

Since I've added so many lists, I was also interested in optimizing/consolidating the number of lists. Apparently the AdGuard DNS already consolidates several other lists.

**Reply by tackynugget** [reply]
Tagging @ameshkov who moved this to discussion. Please note:

> One thing I should have mentioned earlier was that I can reproduce the desired state when I ALSO activate the AdGuard AdBlocker Chrome browser extension. I reviewed the lists the extension used to block the canyoublockit images (those images are the ads, btw--many other forum posts here seem to not understand that), but I couldn't find any difference between the lists the extension used and the lists AdGuardHome uses. Ideally, I'd like my network to enjoy the same level of ad blocking that the chrome extension provides.

**Reply by tackynugget** [reply]
@bbartlomiej I added all of the lists that you had that I didn't have. I still see the ads on canyoublockit.com. When I activate the AdGuard Google Chrome browser plug in, with fewer/overlapping lists, it blocks those ads. I'm just after parity between the plugin and the AdGuardHome DNS filter.

**Reply by ameshkov** [reply]
Unfortunately, DNS-level blocking is kinda limited compared to what full-scale content blockers can do - it can't block ads loaded from the same domain and there's no cosmetic filtering.

**Reply by bbartlomiej** [reply]
> 
> 
> @bbartlomiej I added all of the lists that you had that I didn't have. I still see the ads on canyoublockit.com. When I activate the AdGuard Google Chrome browser plug in, with fewer/overlapping lists, it blocks those ads. I'm just after parity between the plugin and the AdGuardHome DNS filter.

If you still see ads you must add the domains to your lists. It is not possible to filter ads server from the same domain as the genuine content.

**Reply by tackynugget** [reply]
Thanks for setting the expectation here @ameshkov 

I was confused by @bbartlomiej comment 
> This depends heavily on the blocklists you use. I tried your canyoublockit.com link and on both basic and extreme test I don't see advertisements. AdBlockHome + uBlock Origin

I thought he was implying his block lists were what led to the successful blocking of those advertisements.

**Comment by tackynugget** [comment, marked as answer]
I'm not trying to credit myself, but I do want to mark a solution for this thread with a summary:

@ameshkov clarified the expectation that even though the Google Chrome AdGuard plugin shares some lists with the AdGuardHome DNS service, there is no parity between the two. To pass the canyoublockit tests, you'll also need the plugin. 

@bbartlomiej shared a lot of other cool lists, but they, in themselves, will not pass the canyoublockit tests. Granted, canyoublockit is just one of many arbitrary adblock testers and your mileage may vary based on what lists and supplemental adblocking utilities are used.
